packages/ui/src/Slider/index.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { useState, useRef } from 'react';
+import styled from 'styled-components';
+import { detail, detailTechnical } from '../utils/typography';
+import { theme } from '../PenumbraUIProvider/theme';
+
+interface SliderProps {
+  min?: number;
+  max?: number;
+  step?: number;
+  defaultValue?: number;
+  onChange?: (value: number) => void;
+  leftLabel?: string;
+  rightLabel?: string;
+  showValue?: boolean;
+  valueDetails?: string;
+  showTrackGaps?: boolean;
+  trackGapBackground?: string;
+  showFill?: boolean;
+  fontSize?: string;
+}
+
+const THUMB_SIZE = '16px';
+const TRACK_HEIGHT = '4px';
+const TRACK_GAP_WIDTH = '4px';
```

Comment:
todo (blocking): use `${props => props.theme.spacing(4)}` to get the correct value for any spacing property in css. Sam tries to use all theme values everywhere in Figma (see the screenshot), so should we

<img width="697" alt="image" src="https://github.com/user-attachments/assets/a1c21071-fc06-4a05-8f91-98c88161f4f6">